File: app/routes/conversation.py
"""对话管理：/api/conversations/*。"""

import os

from fastapi import APIRouter, Header, Request

from .. import store
from ..models import ConversationCreateRequest

router = APIRouter()


@router.get("/api/conversations")
async def conversations(request: Request, x_user_id: str = Header(default="")):
    user_id = store.safe_user_id(x_user_id, request)
    return {"user_id": user_id, "conversations": store.list_conversations(user_id)}


@router.post("/api/conversations")
async def create_conversation(payload: ConversationCreateRequest, request: Request, x_user_id: str = Header(default="")):
    user_id = store.safe_user_id(x_user_id, request)
    return {"conversation": store.new_conversation(user_id, payload.title)}


@router.get("/api/conversations/{conversation_id}")
async def get_conversation(conversation_id: str, request: Request, x_user_id: str = Header(default="")):
    user_id = store.safe_user_id(x_user_id, request)
    return {"conversation": store.load_conversation(user_id, conversation_id)}


@router.delete("/api/conversations/{conversation_id}")
async def delete_conversation(conversation_id: str, request: Request, x_user_id: str = Header(default="")):
    user_id = store.safe_user_id(x_user_id, request)
    path = store.conversation_path(user_id, conversation_id)
    if os.path.exists(path):
        os.remove(path)
    return {"ok": True}

File: app-go/internal/handler/conversation.go
package handler

import (
	"net/http"

	"github.com/gin-gonic/gin"

	"infinite-canvas/app-go/internal/store"
)

func (h *Handler) ListConversations(c *gin.Context) {
	userID := store.SafeUserID(c.GetHeader("X-User-Id"), c.ClientIP())
	c.JSON(http.StatusOK, gin.H{
		"user_id":       userID,
		"conversations": h.store.ListConversations(userID),
	})
}

type ConversationCreateRequest struct {
	Title string `json:"title"`
}

func (h *Handler) CreateConversation(c *gin.Context) {
	var payload ConversationCreateRequest
	if err := c.ShouldBindJSON(&payload); err != nil {
		c.JSON(http.StatusBadRequest, gin.H{"detail": "请求参数不正确"})
		return
	}
	userID := store.SafeUserID(c.GetHeader("X-User-Id"), c.ClientIP())
	conversation, err := h.store.NewConversation(userID, payload.Title)
	if err != nil {
		writeStoreError(c, err, "对话不存在")
		return
	}
	c.JSON(http.StatusOK, gin.H{"conversation": conversation})
}

func (h *Handler) GetConversation(c *gin.Context) {
	userID := store.SafeUserID(c.GetHeader("X-User-Id"), c.ClientIP())
	conversation, err := h.store.LoadConversation(userID, c.Param("id"))
	if err != nil {
		writeStoreError(c, err, "对话不存在")
		return
	}
	c.JSON(http.StatusOK, gin.H{"conversation": conversation})
}

func (h *Handler) DeleteConversation(c *gin.Context) {
	userID := store.SafeUserID(c.GetHeader("X-User-Id"), c.ClientIP())
	if err := h.store.DeleteConversation(userID, c.Param("id")); err != nil {
		writeStoreError(c, err, "对话不存在")
		return
	}
	c.JSON(http.StatusOK, gin.H{"ok": true})
}
